packages/vscode-container-client/src/clients/DockerClientBase/DockerClientBase.ts

```diff
@@ -632,13 +615,8 @@ export abstract class DockerClientBase extends ConfigurableClient implements ICo
                 }
 
                 try {
-                    const inspect = JSON.parse(inspectString);
-
-                    if (!isDockerInspectImageRecord(inspect)) {
-                        throw new Error('Invalid image inspect json');
-                    }
-
-                    return [...images, normalizeDockerInspectImageRecord(inspect)];
+                    const inspect = DockerInspectImageRecordSchema.parse(JSON.parse(inspectString));
+                    return [...images, normalizeDockerInspectImageRecord(inspect, inspectString)];

```

Comment:
It might be worth implementing the normalization functions as [zod transforms](https://zod.dev/api?id=transform) to allow the parsing and conversion in a single call: so `return [...images, NormalizedDockerInspectImageRecordSchema.parse(JSON.parse(inspectString))];` could give the correct output.

Reply:
Basically `NormalizedDockerInspectImageRecordSchema` would be `DockerInspectImageRecordSchema.transform(normalizeDockerInspectImageRecord)`.